code/game/objects/effects/decals/decal.dm

```diff
@@ -3,14 +3,19 @@
 	plane = FLOOR_PLANE
 	anchored = TRUE
 	resistance_flags = FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	///Boolean on whether this decal can be placed inside of groundless turfs/walls. If FALSE, will runtime and delete if it happens.
 	var/turf_loc_check = TRUE
 
 /obj/effect/decal/Initialize(mapload)
 	. = ..()
-	if(turf_loc_check && (!isturf(loc) || NeverShouldHaveComeHere(loc)))
+	if(turf_loc_check && NeverShouldHaveComeHere(loc))
+#ifdef UNIT_TESTS
+		stack_trace("[name] spawned in a bad turf ([loc]) at [AREACOORD(src)] in \the [get_area(src)]. Please remove it or set turf_loc_check to FALSE on the decal if intended.")
+#else
```

Comment:
where'd the handling for being in not a turf go

Reply:
I removed it because it would delete the ash in mail, which would fall to the ground when opening them.

Reply:
that seems silly, don't decals like instantly die

Reply:
If it's in mail, it'll wait until it's opened, then drop to the floor, something I fixed in this PR with an ``isitem`` check in ``mail.dm``